crates/oci-distribution/src/client.rs

```diff
@@ -233,8 +247,27 @@ impl Client {
             reqwest::StatusCode::OK => {
                 let digest = digest_header_value(&res)?;
                 let text = res.text().await?;
+                debug!("validating response: {}", text);
+                let versioned: Versioned = serde_json::from_str(&text).with_context(|| {
```

Comment:
This change (these three blocks) looks like it could be pulled out into a new `validate_layer` function which would be more intention-revealing, give a clear home to future validation code and reduce the amount the reader has to process here.